src/portal/ui/viewer/proc_par.cljs

```diff
@@ -0,0 +1,340 @@
+        tree-root     (or (:treeRoot decoded) (get decoded "treeRoot"))
+        tree-children (or (:treeChildren decoded) (get decoded "treeChildren"))
+        _level         (or (:level decoded) (get decoded "level"))
+
+        _coll    [tree-root tree-children]
+        _theme   (theme/use-theme)
+        node-id (flow/useNodeId)]
+
+    ;; NOTE: the data-id of the wrapper changes all the time, causing rerender
+    ;; it is the same as the default though... why?
+    ;; https://reactflow.dev/docs/api/nodes/custom-nodes/
+    ;; https://github.com/wbkd/react-flow/blob/893ee87838ed66ccc7d1db45f68319ec9fe042f2/packages/core/src/components/Nodes/DefaultNode.tsx#L7
+    (r/as-element
+     ^{:key (str "ppar-" node-id)}
+     [:div {:class "proc-par"}
+      ^{:key (str "ppar-target-handle-" node-id)}
+      [flow-handle {:id            (str "ppar-target-handle-" node-id)
+                    :type          "target"
+                    :position      (or (.-targetPosition props) (.-Top Position))
+                    :isConnectable (.-isConnectable props)}]
+      ^{:key (str "ppar-label-" node-id)}
+      [:span {:style {:color (guess-label-color tree-root)}}
+       (.-label data)]
+      [:f> (display-proc-message-body tree-root)]
+      ^{:key (str "ppar-source-handler-" node-id)}
+      [flow-handle {:id            (str "ppar-source-handler-" node-id)
+                    :type          "source"
+                    :position      (or (.-sourcePosition props) (.-Bottom Position))
+                    :isConnectable (.-isConnectable props)}]])))
+
+(def nodeTypes {:procMessageNode custom-reactflow-proc-node})
+
+;;;;
+;; download img button
+;;;;
+
+(defn download-btn
+  "https://reactflow.dev/docs/examples/misc/download-image/
+  https://reactflow.dev/docs/api/plugin-components/controls/"
+  ;; NOTE: this captures the graph as seen - if the graph is to big it will be
+  ;; truncated :/
+  [props]
+  [control-button
+   {:title      "Download as PNG image"
+    :aria-label "Download as PNG image"
+    :on-click   (fn [_]
+                  (-> (html-to-image/toPng (. js/document (querySelector ".react-flow"))
+                                           #js {:filter (fn [node]
+                                                          (let [excluded #{"react-flow__minimap"
+                                                                           "react-flow__controls"}]
+
+                                                            (not (some excluded (.-classList node)))))})
+                      (.then (fn download-img [data-url]
+                               (doto (. js/document (createElement "a"))
+                                 (.setAttribute "download" (str "procparflow_" (:id props) ".png"))
+                                 (.setAttribute "href" data-url)
+                                 (.click))))))}
+   [:div "⇓"]])
+
+;;;;;
+;; rendering
+;;;;;
+
+(def rf-style (io/inline-slurp "./node_modules/reactflow/dist/style.css"))
```

Comment:
I'm noticing this causes issues loading the viewer via sci, I think portal needs a dynamic facility for loading files in viewers 🤔